runtime/shared/shrclssup.c

```diff
@@ -181,7 +181,31 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 #if !defined(WIN32) && !defined(WIN64)
 				/* Get platform default cache directory */
 				rc = j9shmem_getDir(NULL, J9SHMEM_GETDIR_APPEND_BASEDIR, defaultCacheDir, J9SH_MAXPATH);
-				if (-1 == rc) {
+				if (0 > rc) {
+					if (0 != verboseFlags) {
+						switch(rc) {
+						case J9PORT_ERROR_SHMEM_GET_DIR_BUF_OVERFLOW:
+							SHRCLSSUP_ERR_TRACE(verboseFlags, J9NLS_SHRC_GET_DIR_BUF_OVERFLOW);
+							break;
+						case J9PORT_ERROR_SHMEM_GET_DIR_FAILED_TO_GET_HOME:
+							SHRCLSSUP_ERR_TRACE(verboseFlags, J9NLS_SHRC_GET_DIR_FAILED_TO_GET_HOME);
+							break;
+						case J9PORT_ERROR_SHMEM_GET_DIR_HOME_BUF_OVERFLOW:
+							j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_SHRC_GET_DIR_HOME_BUF_OVERFLOW);
```

Comment:
This one using j9nls_printf() is inconsistent with the others. Although they could all use j9nls_printf() since the verboseFlags are checked above.

Reply:
Changed the code to call new helper function j9shr_getCacheDir().